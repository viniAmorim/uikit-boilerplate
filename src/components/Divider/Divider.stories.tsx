import React from 'react';

import { Story, Meta } from '@storybook/react';

import Divider, { DividerProps } from './Divider';

export default {
  title: 'Atoms/Divider',
  component: Divider,
} as Meta;

const Template: Story<DividerProps> = (args) => <Divider {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Green = Template.bind({});
Green.args = {
  color: 'green',
};

export const Width = Template.bind({});
Width.args = {
  w: '50%',
};

export const Vertical = Template.bind({});
Vertical.args = {
  orientation: 'vertical',
};
