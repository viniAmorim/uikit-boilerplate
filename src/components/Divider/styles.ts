import styled, { css } from 'styled-components';

export interface DividerProps {
  color?: 'green' | 'grey';
  w?: string;
  h?: string;
  orientation?: 'vertical' | 'horizontal';
}

const orientationModifier = {
  vertical: css`
    border-left: 1px solid
      ${(props: DividerProps) =>
        props.color === 'green'
          ? ({ theme }) => theme.colors.green500
          : ({ theme }) => theme.colors.grey100};
    border-left-width: 1px;
  `,
  horizontal: css`
    border-bottom: 1px solid
      ${(props: DividerProps) =>
        props.color === 'green'
          ? ({ theme }) => theme.colors.green500
          : ({ theme }) => theme.colors.grey100};
  `,
};

export const Divider = styled.div<DividerProps>`
  width: ${(props) => props.w};
  height: ${(props) => props.h};
  margin: 4px;

  ${({ orientation = 'horizontal' }) => orientationModifier[orientation]}
`;
