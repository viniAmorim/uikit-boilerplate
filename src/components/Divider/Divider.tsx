import React from 'react';

import * as Styled from './styles';

export type DividerProps = Styled.DividerProps;

const Divider = ({
  color = 'grey',
  w = '100%',
  h = '22px',
  orientation = 'horizontal',
  ...rest
}: DividerProps) => {
  return (
    <Styled.Divider
      color={color}
      w={orientation === 'vertical' ? '1px' : w}
      h={orientation === 'horizontal' ? '1px' : h}
      orientation={orientation}
      {...rest}
    />
  );
};

export default Divider;
