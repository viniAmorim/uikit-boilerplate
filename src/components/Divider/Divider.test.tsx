import React from 'react';

import { render } from '~/test/utils';

import Divider from './Divider';

describe('Divider', () => {
  it('should render correctly', () => {
    const { container } = render(<Divider />);
    const divider = container.firstChild as HTMLElement;

    expect(divider.nodeName).toBe('DIV');
    expect(divider).toBeInTheDocument();
    expect(divider).toMatchSnapshot();
  });

  it('should render a vertical divider when orientation is set to vertical', () => {
    const { getByTestId } = render(
      <Divider data-testid="divider" orientation="vertical" />,
    );
    const divider = getByTestId('divider');

    expect(divider).toHaveAttribute('orientation', 'vertical');
  });

  it('should apply green color when color prop is set to green', () => {
    const { getByTestId } = render(
      <Divider data-testid="divider" color="green" />,
    );
    const divider = getByTestId('divider');

    expect(divider).toHaveAttribute('color', 'green');
  });
});
