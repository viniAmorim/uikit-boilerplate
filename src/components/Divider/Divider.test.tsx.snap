// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render correctly 1`] = `
<div
  class="sc-beqWaB dwmPGf"
  color="grey"
  orientation="horizontal"
/>
`;
